themes/rjorgenson/rjorgenson.theme.sh

```diff
@@ -14,88 +14,82 @@ SCM_THEME_PROMPT_CLEAN=" ${bold_green}✓${normal}"
 SCM_GIT_CHAR="${STRING_COLOR}±${normal}"
 SCM_SVN_CHAR="${bold_cyan}⑆${normal}"
 SCM_HG_CHAR="${bold_red}☿${normal}"
+PROMP_CHAR="${BRACKET}➞ ${normal}"
 
 #Mysql Prompt
 export MYSQL_PS1="(\u@\h) [\d]> "
 
+TITLEBAR=""
 case $TERM in
-        xterm*)
-        TITLEBAR="\[\033]0;\w\007\]"
-        ;;
-        *)
-        TITLEBAR=""
-        ;;
+        xterm*) TITLEBAR="\[\033]0;\w\007\]" ;;
 esac
 
-PS3=">> "
+if [[ $OSTYPE =~ "darwin" ]]; then
+    echo "${BRACKET}➞  ${normal}"
+fi
 
 __my_rvm_ruby_version() {
-    local gemset=$(echo $GEM_HOME | awk -F'@' '{print $2}')
-  [ "$gemset" != "" ] && gemset="@$gemset"
-    local version=$(echo $MY_RUBY_HOME | awk -F'-' '{print $2}')
-    local full="$version$gemset"
-  [ "$full" != "" ] && echo "${BRACKET_COLOR}[${STRING_COLOR}$full${BRACKET_COLOR}]${normal}"
+    local gemset ; gemset=$(echo "${GEM_HOME}" | awk -F'@' '{print $2}')
+    local version ; version=$(echo "${MY_RUBY_HOME}" | awk -F'-' '{print $2}')
+    [ "${gemset}" != "" ] && gemset="@${gemset}"
+    local full="${version}${gemset}"
+    [ "${full}" != "" ] && echo "${BRACKET}[${STRING_COLOR}${full}${BRACKET}]${normal}"
 }
 
 is_vim_shell() {
-        if [ ! -z "$VIMRUNTIME" ]
-        then
-                echo "${BRACKET_COLOR}[${STRING_COLOR}vim shell${BRACKET_COLOR}]${normal}"
-        fi
+    if [ ! -z "$VIMRUNTIME" ]
+    then
+        echo "${BRACKET}[${STRING_COLOR}vim shell${BRACKET}]${normal}"
+    fi
 }
 
 function is_integer() { # helper function for todo-txt-count
     [ "$1" -eq "$1" ] > /dev/null 2>&1
         return $?
 }
 
+# XXX do we need/want to integrate with todo.sh? We don't provide it and I
+# can't find a version online that accepts ls as an input
 todo_txt_count() {
     if _omb_util_command_exists todo.sh; then # is todo.sh installed
-        count=$(todo.sh ls | egrep "TODO: [0-9]+ of ([0-9]+) tasks shown" | awk '{ print $4 }')
-        if is_integer $count; then # did we get a sane answer back
-            echo "${BRACKET_COLOR}[${STRING_COLOR}T:$count${BRACKET_COLOR}]$normal"
+        count=$(todo.sh ls \
+          | awk '/TODO: [0-9]+ of ([0-9]+) tasks shown/ { print $4 }')
+        if [ "${count}" -eq "${count}" ] &> /dev/null
```

Comment:
Why don't you use `is_integer`? I haven't tested it, but is there any problem with using `is_integer`? If there is any, we should rather fix `is_integer` instead of inlining the function body of `is_integer`.

Reply:
Why check the return code of a single use function when we can just compare the input to itself.

I don't see a reason to keep it around unless we need to check in multiple places.

Reply:
Because the intent of `[ "${count}" -eq "${count}" ] &> /dev/null` is not so obvious. We might have added a comment here if necessary, but if we can *name* the code fragment by making it a function, that's better than a comment.

Reply:
I still don't think it's the best idea, unless we use the function in more places, but I added the function back in.

Reply:
Maybe the existing approach is not the best approach, but I don't think directly writing this rarely used trick inline is the *best idea* either. Then, I don't see any reason to move from one not-best approach to another.